content/kb/troubleshooting-restore-pvc-stuck-using-velero-csi-plugin-version-lower-than-0.4.md

```diff
@@ -0,0 +1,30 @@
+---
+title: "Troubleshooting: Restore PVC stuck using Velero CSI plugin version < 0.4"
+author: Ray Chang
+draft: false
+date: 2022-12-15
+categories:
+  - "restore"
+---
+
+## Applicable versions
+
+All Longhorn versions.
+
+## Symptoms
+
+PersistentVolumeClaim (PVC) stuck in `Pending` state after restore with Velero < 0.4.
+
+## Reason
+
+When using Velero CSI plugin version < 0.4 to restore a PVC, only the GA storage-provisioner `volume.kubernetes.io/storage-provisioner` in the annotation will be restored, but without the beta version `volume.beta.kubernetes.io/storage-provisioner`. 
+
+Longhorn finds the beta version of storage-provisioner in annotation and provision the volume.
```

Comment:
```suggestion
For Longhorn versions using `longhornio/csi-provisioner:v2.1.2`, when processing a PVC to provision the volume, Longhorn CSI provisioner will only recognize the `volume.beta.kubernetes.io/storage-provisioner` annotation which will be tagged to each PVC via Kubernetes after determining the storage provisioner. This annotation of the PVC will be backed up together via Velero.

After restoring the PVC via Velero with its CSI plugin (< 0.4), it will only remove the `volume.beta.kubernetes.io/storage-provisioner` but keep the `volume.kubernetes.io/storage-provisioner` annotation intact of the PVC, because the plugin doesn't respect the general available `volume.kubernetes.io/storage-provisioner` annotation. Thus, it will cause the restoring PVC will be failed to be processed by the built-in Longhorn CSI provisioner and be stuck in the `Pending` state.

This compatibility issue is caused by the Velero CSI plugin and it has been fixed in the following versions, so since the 0.4 version, all annotations will be respected to ensure the corresponding volume provision is correct.
```